lib/model/typing_status.dart

```diff
@@ -64,7 +64,7 @@ class TypingStatus extends ChangeNotifier {
   void handleTypingEvent(TypingEvent event) {
     SendableNarrow narrow = switch (event.messageType) {
       MessageType.direct => DmNarrow(
-        allRecipientIds: event.recipientIds!..sort(), selfUserId: selfUserId),
+        allRecipientIds: event.recipientIds!.toList()..sort(), selfUserId: selfUserId),
```

Comment:
No need for the copy (i.e. this `toList` call). In principle it's not ideal to be mutating the list because in principle something else could be looking at it… but (a) nothing else actually does look at this, because it's a specific feature that this model is entirely responsible for, and (b) really it ought to be sorted anyway, so this is just doing a public service for anything else that might be using the data.

(The ideal version of this would be that we sort at deserialization time.)

Reply:
The DmNarrow.ofMessage factory creates a unmodifiable list, which motivates the toList call here, because we can't mutate the list.

Reply:
I think we might not be using DmNarrow.ofMessage in the application code, so tweaking the tests can be a solution. But it would be helpful if there is a way to statically discover issues like this.

Reply:
Resolved by only sorting them when deserializing TypingEvent, and skipping the sort call from TypingStatus.